tests/contrib/kafka/test_kafka_dsm.py

```
@@ -222,10 +222,24 @@ def _read_single_message(consumer):
     assert len(buckets) == 1
     assert list(buckets.values())[0].latest_produce_offsets[PartitionKey(kafka_topic, 0)] > 0
 
+    def _wait_for_auto_commit_and_fetch_offset(timeout=5.0):
+        start_time = time.time()
+
+        while time.time() - start_time < timeout:
+            tp = TopicPartition(kafka_topic, 0)
+            committed = consumer.committed([tp], timeout=1.0)
+
+            # Check for valid committed offset (> 0, not -1001/_NO_OFFSET)
+            if committed and committed[0].offset > 0:
+                return committed[0].offset
+
+            time.sleep(0.1)
+
+        return None
+
     # Auto commit is enabled so we want to wait for the commit event to fire
-    time.sleep(1)
-    first_offset = consumer.committed([TopicPartition(kafka_topic, 0)])[0].offset
-    if first_offset:
+    first_offset = _wait_for_auto_commit_and_fetch_offset()
+    if first_offset is not None:
```

This feels weird to only do the assert if the offset is not none? Essentially we're silently skipping the test behavior in that case. 

Instead can we validate that the first offset is not none in the 5 sec poll window. So instead of 
```
if first_offset is not None:
...
```
we can have
```
assert first_offset is not None, "Auto-commit did not complete within 5 seconds"
...
```

---

I added the assert and removed the if